doc/specification/language-proposal.md

```diff
@@ -0,0 +1,1518 @@
+
+A qualified identifier of the form α._id_, where α is a (qualified) identifier and _id_ is an identifier, is bound to the referent of _id_ in the scope determined by α:
+
+1. In an expression context, if α is bound to a record value, then α._id_ is bound to the value of the field named _id_ in α.
+3. In an expression context, if α is bound to an imported module, then α._id_ is bound to the constant, parameter, or function bound to _id_ in the module referred to by α.
+2. In a type context, if α is bound to an imported module, then α._id_ is bound to the type bound to _id_ in in the module referred to by α.
+3. In a region context, if α is bound to a regime, then α._id_ is bound to the inner regime named _id_ in α.
+
+### Examples
+
+```
+# Binds 'foo' in module context, global scope.
+module foo {
+    # Binds 't' in expression context, module scope.
+    def t: time = 10 s;
+}
+
+module bar {
+    # Binds 'foo' in type context, module scope.
+    type foo = length;
+
+    # Binds 'foo' in expression context, module scope.
+    def foo: foo = 3 m;
+
+    # Refers to 'foo' in module context, from outer global scope.
+    # Binds 'F' in expression context, module scope.
+    import foo as F;
+
+    # Binds 'quux' in expression context, module scope.
+    def quux =
+        # 'let' rebinds 'F' in the subsequent expression context in
+        # expression scope.
+        #
+        # On the right hand side, 'F' is still bound to the module import,
+        # and 'u', a record field name, is bound in record context in the
+        # scope of the record value construct.
+        #
+        # 'F.t' is a qualified identifier, that is bound to the constant 't'
+        # defined in the module 'foo'.
+        let F = { u = F.t; };
+
+        # 'F' refers to the record above; 'F.u' is a qualified identifier
+        # that refers to the value bound to the field 'u' in 'F'.
+        F.u / 20 s;
+
+    # Error: foo cannot refer to both a value and an imported module in module scope.
+    import foo;
+
+    # Error: foo cannot refer to both a value and a function in module scope.
+    def foo = fn () → 4;
+}
+```
+
+## Types and literals
+<a id="types-and-literals"/>
+
+Every expression has a type, which is either:
+
+1. Boolean.
+2. A quantity.
+3. A record type, comprising an unordered sequence of named fields, with each field being either a quantity or another record type.
+4. A function.
+
+A type expression _type-expr_ identifies a particular type, and is used in type assertions and function argument specifiers. Function types have no corresponding type expression: expressions of a function type may only be bound to an identifier or used in a function application expression. In particular, they may not be used as an argument in a function application.
+
+A _type-expr_ is either `boolean`, a quantity term, a record type description, or a (qualified) identifier bound to a type expression by a [type alias](#type-aliases).
```

Comment:
We had a `string` literal above, do we need/want a string type?

Reply:
I don't think we do, as they aren't things that can be passed around or bound to a variable at any point.